include/openPMD/IO/AbstractIOHandlerImpl.hpp

```diff
@@ -303,6 +306,28 @@ class AbstractIOHandlerImpl
    * The region of the chunk should be written to physical storage after the operation completes successfully.
    */
   virtual void writeDataset(Writable*, Parameter< Operation::WRITE_DATASET > const&) = 0;
+
+  /** Get a view into a dataset buffer that can be filled by a user.
+   *
+   * The operation should fail if m_handler->m_frontendAccess is Access::READ_ONLY.
+   * The dataset should be associated with the Writable.
+   * The operation should fail if the dataset does not exist.
+   * The operation should fail if the chunk extent parameters.extent is not smaller or equals in every dimension.
+   * The operation should fail if chunk positions parameters.offset+parameters.extent do not reside inside the dataset.
+   * The dataset should match the dataype parameters.dtype.
+   * The buffer should be stored as a cast-to-char pointer to a flattened version of the backend buffer in parameters.out->ptr. The chunk is stored row-major.
+   * The buffer's content should be written to storage not before the next call to AbstractIOHandler::flush where AbstractIOHandler::m_flushLevel == FlushLevel::InternalFlush.
+   * The precise time of data consumption is defined by the backend:
+   * * Data written to the returned buffer should be consumed not earlier than the next call to AbstractIOHandler::flush where AbstractIOHandler::m_flushLevel == FlushLevel::InternalFlush.
+   * * Data should be consumed not later than the next Operation::ADVANCE task where parameter.mode == AdvanceMode::ENDSTEP.
+   *
+   * This task is optional and should either (1) not be implemented by a backend at all or (2) be implemented as indicated above and set parameters.out->taskSupportedByBackend = true.
```

Comment:
I am not sure if "task" is the right naming here. Because of our `IOTasks`? I think they are a bit of an implementation detail and we give this naming here out to the user.

I think we mean "backend has explicit support for buffer creation" - maybe for the variable the name "backendManagedBuffer = true/false" is better?

Reply:
> I think they are a bit of an implementation detail and we give this naming here out to the user.

This documentation is for the methods of `AbstractIOHandlerImpl` – a class that users do not deal with. This file is not even included anywhere in our frontend. So I think, documenting in terms of implementation details should be fine here?

> I think we mean "backend has explicit support for buffer creation" - maybe for the variable the name "backendManagedBuffer = true/false" is better?

I can rename the variable, this name sounds good

Reply:
Yes, my comment was more about the user-facing vaariable name.

Can we add IO before "task" to make the relation clearer? :)
```suggestion
   * This IOTask is optional and should either (1) not be implemented by a backend at all or (2) be implemented as indicated above and set parameters.out->backendManagedBuffer = true.
```